modules/geolocation/geolocation.py

```diff
@@ -129,3 +130,71 @@ def calculate_pixel_to_geo_mapping(self):
 
         # Return matrix product of mappedGeoMatrix and sourcePixelMatrixInverse
         return (mappedGeoMatrix.dot(sourcePixelMatrixInverse))
+    
+class OutputAnalysis:
+
+    def get_best_location(self,inputLocationTupleList):
+
+    # Splits the 3D numpy arrray into three separate arrays
+
+        try:
+            coordPair = np.vstack(inputLocationTupleList[:, 0]).astype(np.float64)
+            errorArray = np.vstack(inputLocationTupleList[:, 1]).astype(np.float64)
+            confidenceArray = np.vstack(inputLocationTupleList[:, 2]).astype(np.float64)
+
```

Comment:
Sorry if this is an uninformed question, but where are the values for errorArray and confidenceArray coming from? Can you give me an idea of what inputLocationTupleList looks like?

Reply:
An example of how the data would look= np.array(
[[10.4, 20.5], [2.4], [0.78]],
[[11.2, 19.9], [2.8], [0.69]],
…
[[10.1, 20.3], [1.9], [0.94]])